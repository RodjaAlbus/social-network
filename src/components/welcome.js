import { onNavigate } from '../main.js'
import {
  auth, updateProfile, createUserWithEmailAndPassword, db, doc, collection, addDoc
} from '../importsFirebase.js'

export const Welcome = () => {
  const div = document.createElement('div')

  const title = document.createElement('h1')
  title.textContent = 'The prank Society'
  const paperEffect = document.createElement('div')
  paperEffect.id = 'paperH1'

  const section = document.createElement('section')
  section.className = 'createPrankster'

  const pranksterName = document.createElement('input')
  pranksterName.setAttribute('type', 'text')
  pranksterName.placeholder = 'Name your prankster'
  pranksterName.id = 'name'
  pranksterName.className = 'inputs'
  pranksterName.maxLength = 5
  //pranksterName.autocomplete = 'off'

  const canvas = document.createElement('canvas')
  canvas.width = 102
  canvas.height = 102
  const ctx = canvas.getContext('2d')
  const image = document.createElement('img')
  image.src = '../img/player/characters.png'
  image.onload = function () {
    canvas.style.backgroundColor = 'rgb(255,255,255)'
    canvas.style.border = '5px solid grey'
    ctx.imageSmoothingEnabled = false
    ctx.drawImage(image, 16, -1, 16, 16, 0, 0, canvas.width, canvas.height)
  }

  const colorButtons = document.createElement('div')
  colorButtons.className = 'colorButtons'
  let theColorOfTheButton = 'blue'

  const btnBlue = document.createElement('button')
  btnBlue.id = 'blue'
  btnBlue.addEventListener('click', () => {
    theColorOfTheButton = '#80d1f9'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, 0, 16, 16, 0, 0, canvas.width, canvas.height)
  })
  const btnPink = document.createElement('button')
  btnPink.id = 'pink'
  btnPink.addEventListener('click', () => {
    theColorOfTheButton = '#f090d1'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, 16, 16, 16, 0, 0, canvas.width, canvas.height)
  })
  const btnOrange = document.createElement('button')
  btnOrange.id = 'orange'
  btnOrange.addEventListener('click', () => {
    theColorOfTheButton = '#ffa492'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, (16 * 2), 16, 16, 0, 0, canvas.width, canvas.height)
  })
  const btnYellow = document.createElement('button')
  btnYellow.id = 'yellow'
  btnYellow.addEventListener('click', () => {
    theColorOfTheButton = '#ffec9c'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, (16 * 3), 16, 16, 0, 0, canvas.width, canvas.height)
  })
  const btnGreen = document.createElement('button')
  btnGreen.id = 'green'
  btnGreen.addEventListener('click', () => {
    theColorOfTheButton = '#b7ff81'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, (16 * 4), 16, 16, 0, 0, canvas.width, canvas.height)
  })
  const btnPurple = document.createElement('button')
  btnPurple.id = 'purple'
  btnPurple.addEventListener('click', () => {
    theColorOfTheButton = '#b18aff'
    ctx.clearRect(0, 0, canvas.width, canvas.height)
    ctx.drawImage(image, 16, (16 * 5), 16, 16, 0, 0, canvas.width, canvas.height)
  })

  const email = document.createElement('input')
  email.setAttribute('type', 'text')
  email.placeholder = 'Email'
  email.id = 'email'
  email.className = 'inputs'
  //email.autocomplete = 'off'
  const password = document.createElement('input')
  password.setAttribute('type', 'text')
  password.placeholder = "Society's Passcode"
  password.id = 'password'
  password.className = 'inputs'
  //password.autocomplete = 'off'
  const alert = document.createElement('p')
  alert.id = 'alert'
  const btnEnter = document.createElement('button')
  btnEnter.textContent = 'Get In Looser :p'
  btnEnter.id = 'buttonEnter'
  btnEnter.className = 'buttons'
  btnEnter.addEventListener('click', () => {
    if (password.value && email.value && pranksterName.value) {
      createUserWithEmailAndPassword(auth, email.value.toLowerCase(), password.value)
        .then(() => {
          // Salvar los datos del usuario (Firebase Store)
          addDoc(collection(db, 'Pranksters'), {
            color: theColorOfTheButton,
            signInEmail: email.value.toLowerCase(),
            userID: auth.currentUser.uid
          })
            .then((data) => {
              console.log(data)
              //console.log(data._key.path.segments[1])
              onNavigate('/message')
            })
            .catch((e) => {
              alert(e.message)
            })
          updateProfile(auth.currentUser, {
            displayName: pranksterName.value
          })
        })
        .catch((error) => {
          // const errorCode = error.code;
          if (password.value.length <= 6 && password.value) {
            alert.textContent = 'Your password must have more than 6 characters'
          } else { alert.textContent = error.message }
        })
    } else { alert.textContent = 'Please complete all the fields' }
  })
  const btnReturningPrankster = document.createElement('button')
  btnReturningPrankster.textContent = 'Have a prankster already?'
  btnReturningPrankster.className = 'footer'
  btnReturningPrankster.addEventListener('click', () => onNavigate('/returningPrankster'))
  const footerPaperEffact = document.createElement('div')
  footerPaperEffact.id = 'footerPaper'

  colorButtons.append(btnBlue, btnPink, btnOrange, btnYellow, btnGreen, btnPurple)
  section.append(pranksterName, canvas, colorButtons, email, password, alert, btnEnter)
  div.append(paperEffect, title, section, footerPaperEffact, btnReturningPrankster)

  return div
}
