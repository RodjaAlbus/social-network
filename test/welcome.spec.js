import { Welcome } from '../src/components/welcome.js'
import 'jest-canvas-mock';

jest.mock('../src/importsFirebase.js')

describe('Welcome', () => {

  beforeEach(()=>{
    document.body.innerHTML = ''

    document.body.innerHTML = '<div id="root"></div>'
    document.body.innerHTML += '<div id="Player"></div>'
    document.body.innerHTML += '<div id="gameArea"></div>'
    document.body.innerHTML += '<div id="roomVertical"></div>'
    document.body.innerHTML += '<div id="roomHorizontal"></div>'

  })

  it('shoud be a function', () => {
    expect(typeof Welcome).toBe('function')
  })

  it('it shoud render correctly', () => {
    //AAA: Arrange Act Assert

    //Arrange:
    
    //Act:
    document.getElementById('root').appendChild(Welcome())
  
    //Assert:
    expect(document.body.innerHTML).toMatchSnapshot()
  })

  it('should login successfully', ()=>{
    document.getElementById('root').appendChild(Welcome())
    document.getElementById('name').value = 'rosalba'
    document.getElementById('email').value = 'rosalbamusician@gmail.com'
    document.getElementById('password').value = '12341234'
    document.getElementById('buttonEnter').click()
    //navigate
    //regect paths
  })


})
