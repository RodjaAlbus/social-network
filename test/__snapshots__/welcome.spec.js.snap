// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Welcome it shoud render correctly 1`] = `"<div id=\\"root\\"><div><div id=\\"paperH1\\"></div><h1>The prank Society</h1><section class=\\"createPrankster\\"><input type=\\"text\\" placeholder=\\"Name your prankster\\" id=\\"name\\" class=\\"inputs\\" autocomplete=\\"off\\"><canvas width=\\"102\\" height=\\"102\\"></canvas><div class=\\"colorButtons\\"><button id=\\"blue\\"></button><button id=\\"pink\\"></button><button id=\\"orange\\"></button><button id=\\"yellow\\"></button><button id=\\"green\\"></button><button id=\\"purple\\"></button></div><input type=\\"text\\" placeholder=\\"Email\\" id=\\"email\\" class=\\"inputs\\" autocomplete=\\"off\\"><input type=\\"text\\" placeholder=\\"Society's Passcode\\" id=\\"password\\" class=\\"inputs\\" autocomplete=\\"off\\"><p id=\\"alert\\"></p><button id=\\"buttonEnter\\" class=\\"buttons\\">Get In Looser :p</button></section><div id=\\"footerPaper\\"></div><button class=\\"footer\\">Have a prankster already?</button></div></div><div id=\\"Player\\"></div><div id=\\"gameArea\\"></div><div id=\\"roomVertical\\"></div><div id=\\"roomHorizontal\\"></div>"`;
